### PR TITLE
perf(offload): accelerate thinking phase residency

### PR DESCRIPTION
**Pull request by GACLove**
## Summary

- Keep the understanding branch resident during incremental Think forwards when it fits the existing `fast` VRAM budget; the initial long-prefix forward remains streamed.
- Evict understanding and prefix-only weights before loading the generation branch, including interrupted-run retry and synchronous-offload paths.
- Compute only the final prefix-token logits and bypass the unused LM head when appending fixed Think text.
- Keep the public VRAM modes and ComfyUI workflow inputs unchanged. Non-Think inference continues to use the previous execution path.

## Performance

H100 80 GB, SenseNova-U1.5-8B-MoT-Preview, BF16, effective Flash Attention, batch 1, 2048x2048 output, 50 steps, seed 42. Generation times exclude model loading and are single measured invocations without warmup.

`Peak allocated` is live PyTorch tensor memory. `Peak reserved` is the highest PyTorch allocator-pool size and includes allocated tensors plus cached blocks; it is the more conservative of the two when estimating required card capacity, but still excludes CUDA context and non-PyTorch allocations.

| Task / mode | Generation time | Peak allocated / reserved VRAM |
|---|---:|---:|
| T2I, old `fast --think` | 135.948 s | 17.42 / 17.89 GiB |
| T2I, new `fast --think` | **93.404 s (-31.3%)** | 17.72 / 18.26 GiB |
| Two-reference Edit, `full --think` reference | 96.240 s | 56.51 / 60.07 GiB |
| Two-reference Edit, old `fast --think` | 324.552 s | 27.40 / 44.60 GiB |
| Two-reference Edit, new `fast --think` | **182.621 s (-43.7% vs old fast)** | 41.42 / 44.97 GiB |

For Edit, the new fast path intentionally uses 14.02 GiB more peak allocated memory than the old fast path to avoid repeated Think-branch transfers; peak reserved increases by only 0.37 GiB. Compared with `full`, the new fast path still reduces peak allocated by 15.09 GiB and peak reserved by 15.10 GiB.

### 40.5 GiB explicit-budget validation

The same two-reference Edit case was also run with:

```text
--vram_mode fast
--fast_vram_budget_gib 40.5
--fast_vram_headroom_gib 2.0
--fast_activation_reserve_gib 4.0
--think
--input_max_pixels 4194304
--width 2048 --height 2048
--num_steps 50 --seed 42
```

An explicit `fast_vram_budget_gib` overrides the default `fast_vram_fraction=0.9` residency watermark.

| Generation time | Peak allocated | Peak reserved | Output |
|---:|---:|---:|---|
| **223.143 s** | **35.90 GiB** | **44.60 GiB** | Exact PNG and Think-text hash match |

This was an explicit-budget run on an 80 GB H100, not a physical 45 GB/MIG-card test. The 40.5 GiB value controls residency decisions against projected allocated memory; it is not a hard cap on `reserved` or external `nvidia-smi` usage. A physical 45 GB device remains close to the observed 44.60 GiB reserved watermark and needs additional CUDA-runtime headroom.

All old/new T2I and Edit PNG hashes match exactly; generated Think-text hashes also match exactly.

## Why

`fast` previously retained generation weights during denoising, but every autoregressive Think token still streamed the understanding branch from CPU. The new phase-aware policy begins residency only after the memory-heavy full-prefix forward and releases it before generation weights are loaded.

## Validation

- `44 passed`
- pre-commit: Ruff import/format and ComfyUI workflow formatting passed
- GitNexus change impact: low risk
- adversarial standards/spec review: no remaining blocker or should-fix

### CODE REVIEW

**Review by yl-1993 — APPROVED** [review]
LGTM. The change is well-scoped and the offload phase handling looks consistent across the prefix/denoise transitions. The new Think fast path keeps understanding weights resident only after the long prefix pass and only within the existing VRAM budget, so it preserves the intended safety behavior while avoiding repeated CPU->GPU transfers during incremental Think decoding.

I also like that the logits optimization is isolated behind `_think_prefix_forward`, and `_append_text_tokens_to_cache` now bypasses the LM head entirely for fixed text cache updates.
